sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts

```diff
@@ -58,19 +54,22 @@ export class ConfirmationOfEnrollmentControllerService extends BaseController {
       );
     }
 
-    if (
-      !options?.allowOutsideCOEWindow &&
-      !this.applicationService.withinValidCOEWindow(
-        new Date(disbursementSchedule.disbursementDate),
-      )
-    ) {
-      throw new UnprocessableEntityException(
-        `Confirmation of Enrollment window is greater than ${COE_WINDOW} days`,
-      );
+    if (!options?.allowOutsideCOEApprovalPeriod) {
+      const approvalPeriodStatus =
+        this.disbursementScheduleService.getCOEApprovalPeriodStatus(
+          disbursementSchedule.disbursementDate,
+          disbursementSchedule.studentAssessment.application.currentAssessment
+            .offering.studyEndDate,
+        );
+      if (
+        approvalPeriodStatus !== COEApprovalPeriodStatus.WithinApprovalPeriod
+      ) {
+        throw new UnprocessableEntityException(
+          "The enrolment cannot be confirmed as current date is not within the valid approval period of it.",
```

Comment:
"The enrolment cannot be confirmed as current date is not within the valid approval period. " I guess we can remove the of it